Source/WebCore/platform/calc/CalcOperator.h

```diff
@@ -63,4 +65,239 @@ enum class CalcOperator : uint8_t {
 
 TextStream& operator<<(TextStream&, CalcOperator);
 
+template <typename T, typename Function>
+double evaluateCalcExpression(CalcOperator calcOperator, const Vector<T>& children, Function&& evaluate)
+{
+    auto getNearestMultiples = [](double a, double b) -> std::pair<double, double> {
+        double lowerB = std::floor(a / std::abs(b)) * std::abs(b);
+        double upperB = lowerB + std::abs(b);
+        return std::make_pair(lowerB, upperB);
+    };
+
+    switch (calcOperator) {
+    case CalcOperator::Add: {
+        double sum = 0;
+        for (auto& child : children)
+            sum += evaluate(child);
+        return sum;
+    }
+    case CalcOperator::Subtract:
+        ASSERT(children.size() == 2);
```

Comment:
Not new: I don’t understand why some of these check sizes but others assert. Would be safer to always check and return NaN. Or more efficient to check elsewhere and assert here. But both depending on the operator?

Reply:
I don't have much history on why the code is written this way, Nikos & Simon wrote most of this code, so I can ask them next week.